lib/order.rb

```diff
@@ -0,0 +1,85 @@
+require_relative "sms"
+
+class Order
+  attr_reader :id, :restaurant, :total, :items
+
+  @@orders = []
+
+  def initialize(restaurant)
+    @id = 1 + @@orders.count
+    @@orders.push(self)
+    @restaurant = restaurant
+    @total = 0
+    @items = []
+  end
+
+  def add_item(item, quantity = 1)
+    raise "This item doesn't exist at this restaurant." if get_item(item).nil?
+    increase_items(item, quantity)
+    increase_total(quantity)
+  end
+
+  def remove_item(item)
+    raise "This item is not in this order." if check_order(item).nil?
+    decrease_total(item)
+    decrease_items(item)
+  end
+
+  def check_total
+    puts "Order Summary:"
+    @items.each { |item| puts "#{item.name}: £#{item.price}" }
+    puts "Your total is £#{@total}"
+  end
+
+  def pay(amount)
+    raise "Sorry, you are missing £#{@total - amount}." unless check_bill(amount)
+    complete_order(amount)
+    send_notification
+  end
+
+  def send_notification
+    SMS.new.send_sms
+  end
+
+  private
```

Comment:
I like the fact that you have well named private methods encapsulating much of the internal logic. It makes it easy to understand what is going on.